src/app/CMakeLists.txt

```diff
@@ -75,18 +75,13 @@ set_source_files_properties(
 )
 
 if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-    # substitute @EXECUTABLE@ in dist/mac/Info.plist
-    get_target_property(EXECUTABLE qbt_app OUTPUT_NAME)
-    configure_file(${qBittorrent_SOURCE_DIR}/dist/mac/Info.plist
-        ${qBittorrent_BINARY_DIR}/dist/mac/pregen.plist @ONLY)
-    file(GENERATE
-        OUTPUT ${qBittorrent_BINARY_DIR}/dist/mac/Info.plist
-        INPUT ${qBittorrent_BINARY_DIR}/dist/mac/pregen.plist
-    )
+    # provide variables for substitution in dist/mac/Info.plist
+    get_target_property(EXECUTABLE_NAME qbt_app OUTPUT_NAME)
+    set(MACOSX_DEPLOYMENT_TARGET ${CMAKE_OSX_DEPLOYMENT_TARGET})
     set_target_properties(qbt_app PROPERTIES
         MACOSX_BUNDLE ON
         MACOSX_BUNDLE_BUNDLE_NAME "qBittorrent"
-        MACOSX_BUNDLE_INFO_PLIST ${qBittorrent_BINARY_DIR}/dist/mac/Info.plist
+        MACOSX_BUNDLE_INFO_PLIST ${qBittorrent_SOURCE_DIR}/dist/mac/Info.plist
```

Comment:
Doesn't this mean that the `Info.plist` file will be changed in the source tree? In addition to substitutions, `configure_file` also copied the file to the binary dir. So I would think that if you're removing the `configure_file` call, you still need to copy the file into the binary dir and revert this change back to `qBittorrent_BINARY_DIR`.

You can use:

```cmake
configure_file(
    ${qBittorrent_SOURCE_DIR}/dist/mac/Info.plist
    ${qBittorrent_BINARY_DIR}/dist/mac/Info.plist
    COPYONLY
)
```

to copy the file. The reason to prefer it to `file(COPY)` is because it is more flexible for when we drop `qmake` (for example, we may wish to introduce some new `@` substitutions or whatever).

Reply:
no, the source file is NOT modified. cmake modifies a copy or does necessary modifications "on the fly". in any case source file remains untouched, only destination file contains values instead of placeholders. tested several times on my own projects and even at work (I had to create macOS .framework using cmake). see attached screenshot:
<img width="936" alt="Screen Shot 2021-04-23 at 20 53 47" src="https://user-images.githubusercontent.com/947647/115910878-0dd39b80-a476-11eb-830d-4012f7b67c52.png">
so, passing the source file completely safe

Reply:
Yeah, I had a look at the CMake source code, and indeed it looks like it calls `ConfigureFile` internally, which makes sense.